dbt/macros/tests/test_format_additional_select_columns.sql

```diff
@@ -0,0 +1,83 @@
+{% macro test_format_additional_select_columns() %}
+    {% do test_format_additional_select_columns_string_element() %}
+    {% do test_format_additional_select_columns_string_list() %}
+    {% do test_format_additional_select_columns_dict_element_no_column() %}
+    {% do test_format_additional_select_columns_dict_element_no_alias() %}
+    {% do test_format_additional_select_columns_dict_element_with_alias() %}
+    {% do test_format_additional_select_columns_dict_element_array_agg() %}
+    {% do test_format_additional_select_columns_dict_element_array_w_alias() %}
+{% endmacro %}
+
+{% macro test_format_additional_select_columns_string_element() %}
+    {{
+        assert_equals(
+            "test_format_additional_select_columns_string_element",
+            format_additional_select_columns(["foo"]),
+            "foo",
+        )
+    }}
+{% endmacro %}
+
+{% macro test_format_additional_select_columns_string_list() %}
+    {{
+        assert_equals(
+            "test_format_additional_select_columns_string_element",
+            format_additional_select_columns(["foo", "bar"]),
+            "foo,bar",
+        )
+    }}
+{% endmacro %}
+
+{% macro test_format_additional_select_columns_dict_element_no_column() %}
+    {{
+        assert_equals(
+            "test_format_additional_select_columns_dict_element_no_column",
+            _format_additional_select_columns(
+                [{"alias": "foo"}], mock_raise_compiler_error
+            ),
+            "Missing required \"column\" key in config: {'alias': 'foo'}",
+        )
+    }}
+{% endmacro %}
+
+{% macro test_format_additional_select_columns_dict_element_no_alias() %}
+    {{
+        assert_equals(
+            "test_format_additional_select_columns_dict_element_no_alias",
+            format_additional_select_columns([{"column": "foo"}]),
+            "foo",
+        )
+    }}
+{% endmacro %}
+
+{% macro test_format_additional_select_columns_dict_element_with_alias() %}
+    {{
+        assert_equals(
+            "test_format_additional_select_columns_dict_element_with_alias",
+            format_additional_select_columns([{"column": "foo", "alias": "bar"}]),
+            "foo as bar",
+        )
+    }}
+{% endmacro %}
+
+{% macro test_format_additional_select_columns_dict_element_array_agg() %}
+    {{
+        assert_equals(
+            "test_format_additional_select_columns_dict_element_array_agg",
+            format_additional_select_columns([{"column": "foo", "agg_func": "max"}]),
+            "max (foo) as foo",
```

Comment:
**suggestion (non-blocking):** I'm not sure that I like this as default behavior since it's not how SQL behaves in most dialects. IMO, we should raise an exception if there's an aggregation function and no alias. Open this behavior if you think it's more flexible though!

Reply:
Could you say more about what you mean by this not being how SQL behaves in most dialects? My thinking is that it's convenient to have `alias` default to `column` since in most of the cases where we're using it we don't need an alias that's different from the column name, but if there's something about that behavior that's confusing or contrary to the SQL standard I'm happy to raise an error instead.

Reply:
Something like:

```sql
SELECT MAX(major_class_code)
FROM ccao.class_dict
```

will return a column named `_col0` in Athena or `max(major_class_code)` in other SQL dialects. In other words, it _doesn't_ return the column name that is input to the aggregation function, as this macro does. IMO the the macro behavior is slightly confusing/not explicit since it doesn't match SQL.

Reply:
I see, thanks for the clarification! What do you think about defaulting to `{agg_func}_{column}`, e.g. `max_major_class_code` in this case? I think that might more closely match the typical SQL behavior you've described here as opposed to raising an exception.

Reply:
I went ahead and implemented this approach in 763d1ebd1ca827a3ba798152eea4504f55d22198.